zanata-war/src/main/java/org/zanata/rest/editor/service/UserService.java

```diff
@@ -36,27 +44,67 @@
     @In
     private AccountDAO accountDAO;
 
+    @In
+    private PersonDAO personDAO;
+
     @Override
     public Response getMyInfo() {
-        return createUser(authenticatedAccount);
+        User user = generateUser(authenticatedAccount);
+        if (user == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        return Response.ok(user).build();
     }
 
     @Override
     public Response getUserInfo(String username) {
+        User user = generateUser(username);
+        if (user == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        return Response.ok(user).build();
+    }
+
+    @Override
+    public User generateUser(String username) {
+        if(StringUtils.isBlank(username)) {
+            return null;
+        }
         HAccount account = accountDAO.getByUsername(username);
-        return createUser(account);
+        return generateUser(account);
     }
 
-    private Response createUser(HAccount account) {
-        if (account == null) {
-            return Response.status(Response.Status.NOT_FOUND).build();
+    @Override
+    public User generateUser(HAccount account) {
```

Comment:
I would suggest to rename it to ```transform``` or ```transferToUser``` to be consistent with other entity to dto conversion.